sklearn/metrics/tests/test_common.py

```diff
@@ -1762,6 +1762,12 @@ def check_array_api_metric(
             metric_kwargs["sample_weight"], device=device
         )
 
+    multioutput = metric_kwargs.get("multioutput")
+    if multioutput is not None:
+        if isinstance(multioutput, np.ndarray):
+            multioutput = xp.asarray(multioutput, device=device)
+        metric_kwargs["multioutput"] = multioutput
```

Comment:
Would it be possible to simplify, since I think this is the only case where we would need to override?
```suggestion
            metric_kwargs["multioutput"] = xp.asarray(multioutput, device=device)
```

Reply:
Thanks for your comment! 

Unfortunately not, there is also the `multioutput="raw_values"` case.

Reply:
But won't that already be set to "raw_values" inside metric_kwargs["multioutput"]

Reply:
Actually you are right, I did not apply your diff correctly locally I think originally. OK let's try and see what the CI has to say about it!

Reply:
If the tests pass I think we might not even need the `is None` condition. We can simply check if multioutput is an array and update only in that case.

Reply:
Yep exactly the last commit I just pushed when looking at the code in my editor.